### PR TITLE
Add context size picker for Claude Agent models

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Refs #319039

Surfaces the CAPI tiered context window (e.g. 200K default vs 1M long-context) in the Claude Code model picker, and plumbs the user's selection through session state into the language-model server so its `modelMaxPromptTokens` override reflects the chosen tier. CAPI bills the long-context tier automatically based on actual prompt size, so no extra header or request flag is required.

### Changes

- **Schema** ([claudeCodeModels.ts](https://github.com/microsoft/vscode/blob/tyleonha/claude-context-size-picker/extensions/copilot/src/extension/chatSessions/claude/node/claudeCodeModels.ts)): exports `CLAUDE_CONTEXT_SIZE_PROPERTY` and emits a `Context Size` enum in the model picker when the endpoint has tiered CAPI pricing (`tokenPricing.default.contextMax < modelMaxPromptTokens`). Description differs based on whether CAPI advertises a long-context surcharge ("Longer sessions" vs "Longer sessions without compaction"), mirroring the patterns in the Copilot LM provider and Copilot CLI session.
- **Session state** (`common/claudeSessionStateService.ts` + `node/claudeSessionStateService.ts`): adds `contextSize: number | undefined` to `SessionState` and a `get/setContextSizeForSession` pair, preserved across all existing setters.
- **Write path** (`claudeChatSessionContentProvider.ts`): reads `request.modelConfiguration[CLAUDE_CONTEXT_SIZE_PROPERTY]` per request and stores it via `setContextSizeForSession`, mirroring the existing reasoning-effort flow.
- **Read path** (`claudeLanguageModelServer.ts`): replaces the hardcoded `DEFAULT_MAX_TOKENS - DEFAULT_MAX_OUTPUT_TOKENS` override with `Math.max(defaultPromptBudget, sessionContextSize - DEFAULT_MAX_OUTPUT_TOKENS)` when the session has a selection.
- **Context usage widget**: no changes needed — it already reads `modelConfiguration?.contextSize` directly, so the usage bar reflects the selected tier automatically.

### Open question for review

The current schema sets `default = defaultContextMax` (the smaller tier). The linked issue argues we should default to the full max instead, since for models without a long-context surcharge the user gains capability with no billing downside. Happy to flip the default if that's the preferred behavior — it's a one-line change in `buildConfigurationSchema`.

### Tests

12 new tests across [`claudeCodeModels.spec.ts`](https://github.com/microsoft/vscode/blob/tyleonha/claude-context-size-picker/extensions/copilot/src/extension/chatSessions/claude/node/test/claudeCodeModels.spec.ts) and [`claudeSessionStateService.spec.ts`](https://github.com/microsoft/vscode/blob/tyleonha/claude-context-size-picker/extensions/copilot/src/extension/chatSessions/claude/node/test/claudeSessionStateService.spec.ts) covering schema emission, omission edge cases, surcharge-vs-no-surcharge labeling, coexistence with `reasoningEffort`, and the new session-state getter/setter pair. All 116 tests in the affected suites pass.

### CODE REVIEW

File: extensions/copilot/src/extension/chatSessions/claude/node/claudeLanguageModelServer.ts [hunk 1]
```diff
@@ -191,6 +191,16 @@ export class ClaudeLanguageModelServer extends Disposable {
...
 
+			// If the user picked a larger context tier in the model picker, raise
+			// the reported prompt token budget so internal accounting and downstream
+			// token-budget computations reflect the chosen tier. CAPI bills tiers
+			// based on actual prompt size, so no other signaling is required.
+			const sessionContextSize = sessionId ? this.sessionStateService.getContextSizeForSession(sessionId) : undefined;
+			const defaultPromptBudget = DEFAULT_MAX_TOKENS - DEFAULT_MAX_OUTPUT_TOKENS;
+			const modelMaxPromptTokens = sessionContextSize !== undefined
+				? Math.max(defaultPromptBudget, sessionContextSize - DEFAULT_MAX_OUTPUT_TOKENS)
+				: defaultPromptBudget;
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply extensions/copilot/src/extension/chatSessions/claude/node/claudeLanguageModelServer.ts:203]: Good catch — fixed in c6c0fdf. `tokenPricing.default.contextMax` and `endpoint.modelMaxPromptTokens` are both prompt-token limits, matching how the regular Copilot LM provider applies `cloneWithTokenOverride(contextSize)` directly. The server now passes `sessionContextSize` straight through (clamped to today's default budget so it never lowers it).

**Review by dmitrivMS — APPROVED** [review]
